model/listener/CentralSyncLogListener.php

```diff
@@ -122,22 +120,52 @@ private function createSyncLogRecord(SyncRequestEvent $event)
             new DateTime()
         );
 
-        $syncLogService->create($syncLogEntity);
+        return $syncLogService->create($syncLogEntity);
     }
 
     /**
-     * @param SyncFinishedEvent $event
+     * @param AbstractSyncEvent $event
+     * @return SyncLogEntity
+     *
+     * @throws common_exception_Error
      */
-    public function logSyncFinished(SyncFinishedEvent $event)
+    private function getSyncLogEntityForEvent(AbstractSyncEvent $event)
     {
         try {
             $params = $event->getSyncParameters();
             $this->validateParameters($params);
-            $syncLogService = $this->getSyncLogService();
 
-            /** @var SyncLogEntity $syncLogEntity */
-            $syncLogEntity = $syncLogService->getBySyncIdAndBoxId($params[SyncLogServiceInterface::PARAM_SYNC_ID], $params[SyncLogServiceInterface::PARAM_BOX_ID]);
+            return $this->findSyncLogEntity($params[SyncLogServiceInterface::PARAM_SYNC_ID], $params[SyncLogServiceInterface::PARAM_BOX_ID]);
+        } catch (SyncLogEntityNotFound $e) {
+            return $this->createSyncLogRecord($event);
+        }
+    }
+
+    /**
+     * @param string $syncId
+     * @param string $boxId
+     * @return SyncLogEntity
+     *
+     * @throws SyncLogEntityNotFound
```

Comment:
Please add `@throws common_exception_Error` or handle it inside the function.

Reply:
Done